test/test_gen_backend_stubs.py

```diff
@@ -13,6 +14,11 @@
 # The tests here are to confirm that badly formed inputs result in reasonable error messages.
 class TestGenBackendStubs(TestCase):
 
+    def setUp(self) -> None:
+        global _GLOBAL_PARSE_NATIVE_YAML_CACHE
+        _GLOBAL_PARSE_NATIVE_YAML_CACHE.clear()
```

Comment:
@albanD I realized that this test hasn't been running in CI, and it's been failing because of this cache.

I think this came up in a discussion a while back with Ed - the easiest thing for me to do to fix the test is just to clear the cache between runs. I checked and the test takes ~5-10 seconds to run, which isn't too bad.

Reply:
Would using functools.cache avoid this problem?

Reply:
I don't think so - I quickly tried replacing `_GLOBAL_PARSE_NATIVE_YAML_CACHE with a `functools.cache` decorator and still see the failures.